### PR TITLE
Attempts to refactor the get_equipped_item proc

### PR DESCRIPTION
**Pull request by Detective-Google**
## About The Pull Request
Old proc was a giant if statement stack which even I can recognize is probably not good
## Why It's Good For The Game
theoretically reduces overhead

## Changelog
:cl:
refactor:get_equipped_item proc
/:cl:

### CODE REVIEW

File: code/modules/mob/inventory.dm [hunk 1]
```diff
@@ -318,48 +318,32 @@
...
-		items += w_uniform
-	if(include_pockets)
-		if(l_store)
-			items += l_store
-		if(r_store)
-			items += r_store
-		if(s_store)
-			items += s_store
+	if(!include_pockets)
+		items -= list(l_store, r_store, s_store)
```
Comment (Qustinnus) [inline code/modules/mob/inventory.dm:346]: What would happend if these are null? nothing?

Comment (Detective-Google) [reply code/modules/mob/inventory.dm:346]: if there's nothing there just nothing happens, yeah (from my testing at least)

File: code/modules/mob/inventory.dm [hunk 2]
```diff
@@ -318,48 +318,32 @@
...
-		items += back
-	if(head)
-		items += head
-	if(wear_mask)
-		items += wear_mask
-	if(wear_neck)
-		items += wear_neck
+	for(var/obj/item/I in contents)
+		if(I.item_flags & IN_INVENTORY)
+			items += I
```
Comment (Rohesie) [inline code/modules/mob/inventory.dm:332]:
```suggestion
	for(var/obj/item/item_content in contents)
		if(item_content.item_flags & IN_INVENTORY)
			items += item_content
```
Use better var names than single upper-case letters.

Comment (Detective-Google) [reply code/modules/mob/inventory.dm]: Thanks for letting me know I made the critical error of assuming this was correct due to the other procs in the file (big brained move assuming the other code in the file I was trying to fix bad code in was sound.)

**Pull request merged**